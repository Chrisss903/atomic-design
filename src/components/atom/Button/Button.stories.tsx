import Button from "./Button";

export default {
  title: "Button",
  component: Button,
  argTypes: {
    variant: {
      control: "select",
      options: ["text", "outlined", "contained"],
    },
    children: {
      control: "text",
    },
  },
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const DButton = (args: any) => <Button {...args} />;
DButton.args = {
  variant: "contained",
  children: "Click Me",
};

export const Primary = () => <Button variant="text">Primary Button</Button>;

export const Secondary = () => (
  <Button variant="outlined">Secondary Button</Button>
);

export const Tertiary = () => (
  <Button variant="contained">Tertiary Button</Button>
);
