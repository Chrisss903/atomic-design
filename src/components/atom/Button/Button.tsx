import Button from "@mui/material/Button";
import React from "react";

type MyButtonProps = {
  variant: "text" | "outlined" | "contained";
  children: React.ReactNode;
  type?: "button" | "submit" | "reset";
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
};

const MyButton = ({ variant, children, type, onClick }: MyButtonProps) => {
  return (
    <Button type={type} variant={variant} onClick={onClick}>
      {children}
    </Button>
  );
};

export default MyButton;
