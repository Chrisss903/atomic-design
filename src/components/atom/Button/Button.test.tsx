import { render, screen } from "@testing-library/react";
import Button from "./Button";
import "@testing-library/jest-dom";

describe("Button Component", () => {
  test("renders button ", () => {
    const { asFragment } = render(
      <Button variant="contained">Click Me</Button>
    );
    const buttonElement = screen.getByRole("button", { name: /click me/i });
    expect(buttonElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders button with different labels", () => {
    const { asFragment } = render(<Button variant="outlined">Submit</Button>);
    const buttonElement = screen.getByRole("button", { name: /submit/i });
    expect(buttonElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
