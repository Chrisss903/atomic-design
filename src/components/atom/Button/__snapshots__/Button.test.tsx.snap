// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > renders button  1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button Component > renders button with correct label 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button Component > renders button with different labels 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-b4yaxo-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Submit
  </button>
</DocumentFragment>
`;

exports[`Button Component renders button with correct label 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button Component renders button with different labels 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-b4yaxo-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Submit
  </button>
</DocumentFragment>
`;
